contract/main.lua

```diff
@@ -1,15 +1,71 @@
+-- Adjust package.path to include the current directory
+local balances = require '.lib.balances'
+local arns = require '.lib.arns'
+local gar = require '.lib.gar'
+local utils = require '.lib.utils'
 
+local ActionMap = {
+    Transfer = 'Transfer',
+    Vault = 'Vault',
+    BuyRecord = 'BuyRecord',
+    SubmitAuctionBid = 'SubmitAuctionBid',
+    ExtendLease = 'ExtendLease',
+    IncreaseUndernameCount = 'IncreaseUndernameCount',
+    JoinNetwork = 'JoinNetwork',
+    LeaveNetwork = 'LeaveNetwork',
+    IncreaseOperatorStake = 'IncreaseOperatorStake',
+    DecreaseOperatorStake = 'DecreaseOperatorStake',
+    UpdateGatewaySettings = 'UpdateGatewaySettings',
+    SaveObservations = 'SaveObservations'
+}
 
--- Adjust package.path to include the current directory
-package.path = package.path .. ';./lib/?.lua'
-
-local arns = require 'arns'
-local balances = require 'balances'
-local gar = require 'gar'
-
-print(arns.add(1, 2))
-print(arns.subtract(2, 1))
-print(balances.add(1, 2))
-print(balances.subtract(2, 1))
-print(gar.add(1, 2))
-print(gar.subtract(2, 1))
+-- Handlers for contract functions
+Handlers.add(ActionMap.Transfer, utils.hasMatchingTag('Action', ActionMap.Transfer), function(msg)
```

Comment:
this is a pretty long and repetitive chunk. Maybe we can do something like:
```lua
function addAction(Action, fn)
    Handlers.add(Action, utils.hasMatchingTag('Action', Action), fn)
end

-- addAction is a function that adds an action to the Handlers object
addAction(ActionMap.Transfer, balances.transfer)

-- the imported arns module is an object with functions as values
-- we iterate over the function names and add them as actions
local handlerNames = utils.keys(arns)
utils.map(
function (arnsAction)
   addAction(arnsAction, arns[arnsAction])
    end, 
handlerNames
)
```

Reply:
maybe - no need to abstract this soon